applesingle.c

```diff
@@ -915,7 +915,7 @@ static int output_digest(FILE *f, char *buf, unsigned buf_sz, size_t n)
 		n -= r;
 	}
 	MD5_Final((void*)md, &c);
-	printf("%016llx%016llx (md5)\n", md[0], md[1]);
+	printf("%016lx%016lx (md5)\n", md[0], md[1]);
```

Comment:
That change produces a warning when building on 32-bit Linux/x86:
`warning: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'UInt64' {aka 'long long unsigned int'}`
I'm not sure what the right fix is. Perhaps UInt64 should be changed to size_t? No idea what the format string would look like though.

Reply:
huh, thats strange because it was giving me the opposite error before, essentially saying it expected `long long unsigned int` and was only being given `long unsigned int`. Does the type of the values contained in `md` also vary based on if `ENABLE_ENCODING` is defined?

Reply:
The problem arises because you are using a 64-bit compiler but you aren't passing the -m32 flag. I've now committed a change to take care of the printf() format strings for the 64-bit case. But this may or may not be sufficient to achieve compatibility with 64-bit hosts. I haven't tested that.